### ISSUE TITLE
The ListView control does not trigger the horizontal scroll bar

### ISSUE THREAD

**Issue by zxscn** [slint-ui/slint#7799]
### Bug Description

After upgrading from v1.9.2 to v1.10.0, the ListView control no longer triggers the horizontal scroll bar.

Works fine in v1.9.2 :
![Image](https://github.com/user-attachments/assets/48527d31-57b0-4687-84f5-8fb95e460868)

No horizontal in v1.10.0 :
![Image](https://github.com/user-attachments/assets/08f664c5-9aec-47cc-92b6-10f700b8d60f)

### Reproducible Code (if applicable)

```slint
slint::slint!{

    import { ListView, ScrollView} from "std-widgets.slint";

    export component TestApp inherits Window {
        preferred_width: 200px;
        preferred_height: 100px;

        ListView {
            for i in 10 : HorizontalLayout {
                spacing: 5px;
                height: 20px;
                width: 400px;

                Rectangle {
                    background: red.with_alpha(i * 0.1);

                    Text {
                        text: @tr("{}_1", i);
                    }
                }
                Rectangle {
                    background: blue.with_alpha(i * 0.1);

                    Text {
                        text: @tr("{}_2", i);
                    }
                }
            }
        }
    }
}
fn main() {
    TestApp::new().unwrap().run().unwrap();
}
```

### Environment Details

- Slint Version: V1.10.0
- Platform/OS: macOS 15.3.1
- Programming Language: Rust
- Backend/Renderer: Default

**Comment by zxscn** [comment]
The same effect can be achieved by specifying the height and width of the child element

```
import { ListView, ScrollView} from "std-widgets.slint";

export component TestApp inherits Window {
    preferred_width: 200px;
    preferred_height: 100px;

    ListView {
        for i in 10 : HorizontalLayout {
            spacing: 5px;

            Rectangle {
                background: red.with_alpha(i * 0.1);
                height: 20px;
                width: 200px;

                Text {
                    text: @tr("{}_1", i);
                }
            }
            Rectangle {
                background: blue.with_alpha(i * 0.1);
                height: 20px;
                width: 200px;

                Text {
                    text: @tr("{}_2", i);
                }
            }
        }
    }
}
```
